assets/js/modules/analytics-4/datastore/audiences.js

```diff
@@ -203,93 +205,114 @@ const baseActions = {
+				// If there are less than two (MAX_INITIAL_AUDIENCES) configured user audiences, add the Site Kit-created audiences if they exist,
+				// up to the limit of two.
 
-			const siteKitAudiences = availableAudiences.filter(
-				( { audienceType } ) => audienceType === 'SITE_KIT_AUDIENCE'
-			);
+				const siteKitAudiences = availableAudiences.filter(
+					( { audienceType } ) => audienceType === 'SITE_KIT_AUDIENCE'
+				);
 
-			// Audience slugs to sort by:
-			const sortedSlugs = [ 'new-visitors', 'returning-visitors' ];
+				// Audience slugs to sort by:
+				const sortedSlugs = [ 'new-visitors', 'returning-visitors' ];
 
-			const sortedSiteKitAudiences = siteKitAudiences.sort(
-				( audienceA, audienceB ) => {
-					const indexA = sortedSlugs.indexOf(
-						audienceA.audienceSlug
-					);
-					const indexB = sortedSlugs.indexOf(
-						audienceB.audienceSlug
-					);
+				const sortedSiteKitAudiences = siteKitAudiences.sort(
+					( audienceA, audienceB ) => {
+						const indexA = sortedSlugs.indexOf(
+							audienceA.audienceSlug
+						);
+						const indexB = sortedSlugs.indexOf(
+							audienceB.audienceSlug
+						);
 
-					return indexA - indexB;
-				}
-			);
+						return indexA - indexB;
+					}
+				);
 
-			const audienceResourceNames = sortedSiteKitAudiences
-				.slice( 0, MAX_INITIAL_AUDIENCES - configuredAudiences.length )
-				.map( ( { name } ) => name );
+				const audienceResourceNames = sortedSiteKitAudiences
+					.slice(
+						0,
+						MAX_INITIAL_AUDIENCES - configuredAudiences.length
+					)
+					.map( ( { name } ) => name );
 
-			configuredAudiences.push( ...audienceResourceNames );
+				configuredAudiences.push( ...audienceResourceNames );
+			}
 		}
 
 		if ( configuredAudiences.length === 0 ) {
+			const audiencesToCreate = failedSiteKitAudienceResourceNames?.length
+				? failedSiteKitAudienceResourceNames
+				: [ 'new-visitors', 'returning-visitors' ];
+
 			// If there are no configured audiences by this point, create the "new-visitors" and "returning-visitors" audiences,
 			// and add them to the configured audiences.
 			const [ newVisitorsResult, returningVisitorsResult ] =
 				yield commonActions.await(
-					Promise.all( [
-						dispatch( MODULES_ANALYTICS_4 ).createAudience(
-							SITE_KIT_AUDIENCE_DEFINITIONS[ 'new-visitors' ]
-						),
-						dispatch( MODULES_ANALYTICS_4 ).createAudience(
-							SITE_KIT_AUDIENCE_DEFINITIONS[
-								'returning-visitors'
-							]
-						),
-					] )
+					Promise.all(
+						audiencesToCreate.map( ( audienceSlug ) => {
+							return dispatch(
+								MODULES_ANALYTICS_4
+							).createAudience(
+								SITE_KIT_AUDIENCE_DEFINITIONS[ audienceSlug ]
+							);
+						} )
+					)
 				);
 
-			// TODO: Full error handling will be implemented via https://github.com/google/site-kit-wp/issues/8134.
-			if ( ! newVisitorsResult.error ) {
+			const failedAudiencesToRetry = [];
+
+			if ( newVisitorsResult.error ) {
```

Comment:
If we are in the situation of having created just a single audience because it failed the previous time, we are going to need to look up the `audienceResourceName` for the other audience in order to add it to `configuredAudiences`

In other words, say for example we try to create both audiences, "new visitors" is created successfully, but "returning visitors" fails. We then retry and "returning visitors" succeeds. This time around we'll have the newly created "returning visitors" audience to add to `configuredAudiences`, but we also need to look up the "new visitors" audience we created last time to ensure they are both added to `configuredAudiences`.

Reply:
Updated the logic to look up for the newly created audience. Thanks!